javascript2/week1/script.js

```diff
@@ -0,0 +1,123 @@
+};
+
+function displayRecipe(recipe) {
+  const recipeSection = document.getElementById("recipe-section");
+
+  const recipeTitle = document.createElement("h1");
+  recipeTitle.textContent = recipe.title;
+  recipeSection.appendChild(recipeTitle);
+
+  const recipeImage = document.createElement("img");
+  recipeImage.src = recipe.picture_url;
+  recipeImage.alt = recipe.title;
+  recipeSection.appendChild(recipeImage);
+
+  const ingredientsHeading = document.createElement("h2");
+  ingredientsHeading.textContent = "Ingredients";
+  recipeSection.appendChild(ingredientsHeading);
+
+  const ingredientsList = document.createElement("ul");
+  ingredientsList.className = "ingredients";
+
+  for (let ingredient of recipe.ingredients) {
+    const listItem = document.createElement("li");
+    listItem.textContent = `${ingredient.NAME} - ${ingredient.AMOUNT}`;
+    ingredientsList.appendChild(listItem);
+  }
+
+  recipeSection.appendChild(ingredientsList);
+
+  const descriptionHeading = document.createElement("h2");
+  descriptionHeading.textContent = "Description";
+  recipeSection.appendChild(descriptionHeading);
+
+  const descriptionList = document.createElement("ul");
+  descriptionList.className = "description";
+  const descriptionItem = recipe.description.split(".");
+  for (let item of descriptionItem) {
+    const trimmedItem = item.trim();
+    if (trimmedItem) {
+      const listItem = document.createElement("li");
+      listItem.textContent = trimmedItem;
+      descriptionList.appendChild(listItem);
+    }
+  }
+  recipeSection.appendChild(descriptionList);
+}
+const form = document.getElementById("add-recipe-form");
+
+function addNewRecipe(event) {
+  event.preventDefault();
+
+  const title = document.getElementById("recipe-title").value;
+  const picture_url = document.getElementById("recipe-image").value;
+  const ingredientsInput = document.getElementById("recipe-ingredients").value;
+  const description = document.getElementById("recipe-description").value;
+
+  const ingredients = ingredientsInput.split(",").map((item) => {
+    const [name, amount] = item.split("-");
+    return { NAME: name.trim(), AMOUNT: amount.trim() };
+  });
+
+  recipeObject = {
+    id: recipeObject.id + 1,
+    title,
+    picture_url,
+    ingredients,
+    description,
+  };
+
+  displayRecipe(recipeObject);
+  form.reset();
+}
+function searchARecipe() {
+  const searchInput = document.getElementById("search-element");
```

Comment:
Very nice addition! But it looks like it's only called on first run. Should there perhaps be a button to filter the results after the user has typed in the search box? We might need to use `.addEventListener(...)` to wire this up.